### DISCUSSION TITLE
Eval inside $$ blocks

### DISCUSSION THREAD

**Discussion by germanesosa** [jupyter-book/mystmd#746] [General] [answered]
Hi!

Is there a way to include {eval} in latex or math blocks?

Thanks!

**Comment by agoose77** [comment, marked as answer]
Hi @germanesosa,

Right now the answer is no. My feeling is that we probably won't support this; parsing `{eval}` _inside_ LaTeX would require a custom Math parser, which would be a headache. Instead, it would make sense to generate the LaTeX using `{eval}` in a language that's easier to write and debug e.g. Python

```python
from IPython.display import Latex, display

first = "x"
second = "10"

body = f"""
The equation is ${first} + {second}$
"""
math = Latex(body)
```
Then embed it using `eval`

```markdown
The equation is {eval}`math`
```

or as a standalone code cell with referenced output.
